model_interfaces/charlie_interface.py

```diff
@@ -1,143 +1,149 @@
-        self.display_name = settings_dict["display_name"][0]
-        self.initial_costs_usd = settings_dict["usage"]["total_cost"]
-
-        # Update max_tokens
-        headers = {
-            "Content-Type": "application/json",
-            "Cookie": f"session_token={self.token}",
+        self.session = requests.Session()
+        self.login()
+        self.create_chat_tab()
+        self.update_memory_settings()
+        self.update_max_tokens()
+        self.load_initial_settings()
+        self.update_system_prompt(self.system_prompt)
+
+    def login(self):
+        login_url = f"{self.endpoint}/login/"
+        login_data = {"username": self.username, "password": self.password}
+        response = self.session.post(login_url, json=login_data)
+        if response.status_code != 200:
+            raise ValueError("Login failed. Please check your credentials.")
+
+    def update_system_prompt(self, new_system_prompt: str):
+        update_settings_url = f"{self.endpoint}/update_settings/"
+        update_data = {
+            "username": self.username,
+            "category": "system_prompt",
+            "setting": {"system_prompt": new_system_prompt},
         }
-        body = {
-            "username": self.user_name,
+        response = self.session.post(update_settings_url, json=update_data)
+        if response.status_code != 200:
+            raise ValueError("Failed to update system prompt setting.")
+        self.system_prompt = new_system_prompt
+
+    def create_chat_tab(self):
+        create_chat_tab_url = f"{self.endpoint}/create_chat_tab/"
+        chat_tab_data = {
+            "username": self.username,
+            "chat_id": self.chat_id,
+            "chat_name": f"{self.chat_id}",
+        }
+        response = self.session.post(create_chat_tab_url, json=chat_tab_data)
+        if response.status_code != 200:
+            raise ValueError(f"Failed to create chat tab '{self.chat_id}'.")
+
+    def update_max_tokens(self):
+        update_settings_url = f"{self.endpoint}/update_settings/"
+        update_data = {
+            "username": self.username,
             "category": "memory",
-            "setting": "max_tokens",
-            "value": self.max_prompt_size,
+            "setting": {"max_tokens": self.max_prompt_size},
         }
-
-        update = requests.post(
-            self.endpoint + "/update_settings/", headers=headers, json=body
-        )
-
-    def reply(self, user_message: str, agent_response: Optional[str] = None) -> str:
-        headers = {
-            "Content-Type": "application/json",
-            "Cookie": f"session_token={self.token}",
+        response = self.session.post(update_settings_url, json=update_data)
+        if response.status_code != 200:
+            raise ValueError("Failed to update max_tokens setting.")
+
+    # Charlie mnemonic has a lot of memory settings that need to be updated, performance varies greatly based on these settings
```

Comment:
In the future, it would be nice to have a default memory distribution. As a user, I'd just set the max. input size, or the model to use, and the different sections would adjust acordingly.

Reply:
Agree, these settings are probably confusing for the user.